### ISSUE TITLE
configure_file can not replace variables after "\\"

### ISSUE THREAD

**Issue by matyalatte** [mesonbuild/meson#13697]
**Describe the bug**
`confugre_file` can not replace the @@ variables after `\\` with meson 1.5.2.

**To Reproduce**

meson.build
```meson
project('test')

test_conf = configuration_data()
test_conf.set('A', 'str_a')
test_conf.set('B', 'str_b')

configure_file(input : 'test.h.in',
               output : 'test.h',
               configuration : test_conf)
```

test.h.in
```c
const char *TEST_STR = "@A@\\\\@B@";
```

When you type `meson setup build`, meson 1.5.2 replaces only `@A@`.

```c
const char *TEST_STR = "str_a\\@B@";
```

**Expected behavior**
test.h should be like this

```c
const char *TEST_STR = "str_a\\str_b";
```

meson had the expected behavior at least three months ago (maybe 1.4.1)

**system parameters**
* Is this a [cross build](https://mesonbuild.com/Cross-compilation.html) or just a plain native build (for the same computer)?
native build
* what operating system (e.g. MacOS Catalina, Windows 10, CentOS 8.0, Ubuntu 18.04, etc.)
Windows 10
* what Python version are you using e.g. 3.8.0
3.12.2
* what `meson --version`
1.5.2
* what `ninja --version` if it's a Ninja build
not ninja build

**Comment by ghost** [comment]
Hello,

I've also been bitten by this bug and I bisected it down to this commit 410bdf8c6c72fd1a2772e86a2a14298f64f1377b

It seems the commit did it on purpose, so is there any way to replace the variable in the following C definition?
```c
#define WINDOWS_PATH "C:\\\\@DIRECTORY@"
```

Link to the relevant bug report https://github.com/mesonbuild/meson/issues/7165

**Comment by bonzini** [comment]
`#define WINDOWS_PATH "C:\\" "@DIRECTORY@" ` should work?

**Comment by ghost** [comment]
Yes thank you for that, however this smells just like a workaround.

It would have been nice if the fix for Perl didn't regress working C code however.
